src/lib/Libifl/pbsD_submit_resv.c

```diff
@@ -70,9 +70,64 @@ __pbs_submit_resv(int c, struct attropl *attrib, char *extend)
 	struct attropl *pal;
 	int rc;
 	char *ret;
+	svr_conn_t **svr_conns;
+	char job_id_resv[PBS_MAXJOBNAME + 1];
 
-	for (pal = attrib; pal; pal = pal->next)
+	job_id_resv[0] = '\0';
+	for (pal = attrib; pal; pal = pal->next) {
 		pal->op = SET;		/* force operator to SET */
+		if (msvr_mode()) {
```

Comment:
instead of calling this in a loop, call it before the loop and store it in a local var